extensions/git/src/repository.ts

```diff
@@ -999,6 +1000,10 @@ export class Repository implements Disposable {
 		await this.run(Operation.Push, () => this.repository.push(remote, undefined, false, true, forcePushMode));
 	}
 
+	async blame(path: string): Promise<string> {
+		return await this.run(Operation.Blame, () => this.repository.blame(path));
```

Comment:
Should we parse the blame result into something more structured?

Reply:
Similar to `git log`, the output of `git blame` is more or less *free form* once we add more options to it. I thought about returing `string[]` but for `--porcelain`, one info entry for a line can be multiline so I'd like to suggest to leave the parsing work to callers. Does this make sense?